### ISSUE TITLE
Discrepancy in the validation IOU values for Driveable Area segmentation

### ISSUE THREAD

**Issue by linlion0311** [bradyz/cross_view_transformers#50]
I have implemented the author's code, but I noticed a discrepancy in the validation IOU values for road segmentation **(In Paper: Driveable Area = 74.3, My implementation = 68.7).** However, the vehicle category scores are consistent with the scores mentioned in the paper. I would like to ask whether I should consider changing the training labels or how I should proceed with the training. Thank you!

**Comment by flymin** [comment]
I cannot reproduce the results reported in the paper, either. Even with the released checkpoint, the result from validation is only 72.21. Any hint for this?

**Comment by linlion0311** [comment]
> I cannot reproduce the results reported in the paper, either. Even with the released checkpoint, the result from validation is only 72.21. Any hint for this?

Are you using released checkpoint from Github to do validation? Or you using your model

**Comment by flymin** [comment]
> Are you using released checkpoint from Github to do validation? Or you using your model

I use the checkpoint from this repo for validation, and get 72.21 for the driveable area.

**Comment by linlion0311** [comment]
> > Are you using released checkpoint from Github to do validation? Or you using your model
> 
> I use the checkpoint from this repo for validation, and get 72.21 for the driveable area.

Please note your dataset splitting method. I noticed that the author used 'road segment' as the label for drivable areas. After making changes, I achieved an IOU of 73.3.

By the way, could you please share how you use checkpoints to validate the model? I am not very familiar with Python programming. If possible, could you kindly send the complete implementation process to [linlion0311@gmail.com](mailto:linlion0311@gmail.com)? Thank you very much!

**Comment by flymin** [comment]
Hi @linlion0311,

I think in [this line](https://github.com/bradyz/cross_view_transformers/blob/4de6e641397ef1ffde996d7549f7f988e49156f7/config/experiment/cvt_nuscenes_road.yaml#L15), the author defined driveable area as a combination of road segment and lane. Why do you change it to 'road segment'? Is is correct for the settings in the paper?

Sorry I cannot share my source code, you can call `trainer.validate(model_module, val_loader)` similar to [this line](https://github.com/bradyz/cross_view_transformers/blob/4de6e641397ef1ffde996d7549f7f988e49156f7/scripts/train.py#L71)

Update: with the released ckpt and **cvt_labels_nuscenes_v2**, I got **73.69** for road. Even if it is better, it still cannot catch up with **74.3** reported in the paper.

**Comment by linlion0311** [comment]
> 

It's ok, I'll keep studying it. 

Regarding the label segmentation part, I tested using 'lane' + 'road segment', and it converged faster. 
Perhaps this is the reason why the author didn't use 'lane' + 'drivable area' (they are very similar). have a good day!

**Comment by linlion0311** [comment]
Hi @flymin  

Can I ask if you are implementing this repo in a Windows or Ubuntu(Linux) environment?

because I can't implement it in Windows.

**Comment by flymin** [comment]
> Hi @flymin
> 
> Can I ask if you are implementing this repo in a Windows or Ubuntu(Linux) environment?
> 
> because I can't implement it in Windows.

I use Linux. Sorry I cannot help you with Windows.

**Comment by 1146065545** [comment]
> Hi @flymin
> 
> Can I ask if you are implementing this repo in a Windows or Ubuntu(Linux) environment?
> 
> because I can't implement it in Windows.

Hello, could you please tell me how the code provided by the author can test the trained model and how to visualize it? I've been doing this lately as well, thank you very much!

**Comment by flymin** [comment]
Please check the dialogue above. I think you’d better check the doc of pytorch-lightning for testing. And for visualization, I remember the author has provided a notebook.

**Comment by 1146065545** [comment]
> Please check the dialogue above. I think you’d better check the doc of pytorch-lightning for testing. And for visualization, I remember the author has provided a notebook.

Sorry, can you go into more detail, I mean about testing the trained model, does the author provide specific .py files

**Comment by flymin** [comment]
I don’t think so.

**Comment by 1146065545** [comment]
> I don’t think so.

╥﹏╥

**Comment by SunJ1025** [comment]
@flymin hi flymin, I want to ask which **.yaml** did you use to get "**73.69** for road." I only got about 47/48 when I test with the commond `python3 scripts/train.py +experiment=cvt_nuscenes_road data.dataset_dir=/data1/dataset/nuscenes data.labels_dir=/data1/dataset/nuscenes/cvt_labels_nuscenes_v2` 
![download](https://github.com/bradyz/cross_view_transformers/assets/151591258/327db921-da45-410f-b4b3-e916f84b4609)

I test with the released **cvt_nuscenes_road_75k.ckpt** and **cvt_labels_nuscenes_v2**.

**Comment by KimJunHan** [comment]
> @flymin hi flymin, I want to ask which **.yaml** did you use to get "**73.69** for road." I only got about 47/48 when I test with the commond `python3 scripts/train.py +experiment=cvt_nuscenes_road data.dataset_dir=/data1/dataset/nuscenes data.labels_dir=/data1/dataset/nuscenes/cvt_labels_nuscenes_v2` ![download](https://private-user-images.githubusercontent.com/151591258/315904784-327db921-da45-410f-b4b3-e916f84b4609.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0MTk5NDUsIm5iZiI6MTcyMjQxOTY0NSwicGF0aCI6Ii8xNTE1OTEyNTgvMzE1OTA0Nzg0LTMyN2RiOTIxLWRhNDUtNDEwZi1iNGIzLWU5MTZmODRiNDYwOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMVQwOTU0MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMzI2YzdiZjU1NzhlZTZiZWJhZTM3OGIxMGNiMTY3NzJmMWMzNzA4MTYyNGRkYTZmYTMyYTVkNTQ1YTA2NDQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ECLwIY9EA07Zog93_-5vXdpSbcPXZ3fFaMjJ_JW-F9o)
> 
> I test with the released **cvt_nuscenes_road_75k.ckpt** and **cvt_labels_nuscenes_v2**.

**How did you find IOU value like picture ???**

**Comment by zengkaiya** [comment]
> I cannot reproduce the results reported in the paper, either. Even with the released checkpoint, the result from validation is only 72.21. Any hint for this?

Hello, I noticed that MagicDrive also conducted related downstream task enhancement experiments. I would like to ask how they trained the model in the end? Can you share some details about the changes made to the baseline training? Thanks
